packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -58,38 +58,82 @@ type BaseButtonStyleProps = {
   textColor: BaseTextProps['color']; // prop
   spacing: string; // raw value
   text?: string;
+  color: string; // raw value
+  borderColor: string; // raw value
+  hoverColor: string; // raw value
+  hoverBorderColor: string; // raw value
+  activeColor: string; // raw value
+  activeBorderColor: string; // raw value
+  focusColor: string; // raw value
+  focusBorderColor: string; // raw value
+  focusRingColor: string; // raw value
 };
 
 const getProps = ({
-  theme,
-  size,
-  children,
   buttonTypographyTokens,
+  children,
+  isDisabled,
+  size,
+  theme,
 }: {
+  buttonTypographyTokens: typeof typography.onDesktop | typeof typography.onMobile;
+  children?: string;
+  isDisabled: boolean;
   theme: Theme;
   size: BaseButtonCommonProps['size'];
-  children?: string;
-  buttonTypographyTokens: typeof typography.onDesktop | typeof typography.onMobile;
 }): BaseButtonStyleProps => {
   const props: BaseButtonStyleProps = {
     iconSize: 'medium',
     fontSize: buttonTypographyTokens.fonts.size.medium,
-    lineHeight: buttonTypographyTokens.lineHeight.medium,
+    lineHeight: buttonTypographyTokens.lineHeights.medium,
     minHeight: '40px',
     iconSpacing: makeSpace(theme.spacing[2]),
-    iconColor: 'action.icon.primary.default',
-    textColor: 'action.text.primary.default',
+    iconColor: isDisabled ? 'action.icon.primary.disabled' : 'action.icon.primary.default',
+    textColor: isDisabled ? 'action.text.primary.disabled' : 'action.text.primary.default',
     spacing: `${makeSpace(theme.spacing[0])} ${makeSpace(theme.spacing[5])} ${makeSpace(
       theme.spacing[0],
     )} ${makeSpace(theme.spacing[5])}`,
     text: children?.trim(),
+    color: getIn(
+      theme.colors,
+      isDisabled ? 'action.background.primary.disabled' : 'action.background.primary.default',
+    ),
+    borderColor: getIn(
+      theme.colors,
+      isDisabled ? 'action.border.primary.disabled' : 'action.border.primary.default',
+    ),
+    hoverColor: getIn(
+      theme.colors,
+      isDisabled ? 'action.background.primary.disabled' : 'action.background.primary.hover',
+    ),
+    hoverBorderColor: getIn(
+      theme.colors,
+      isDisabled ? 'action.border.primary.disabled' : 'action.border.primary.hover',
+    ),
```

Comment:
I have an urge to refactor this to a function 😢 but not sure if that would be a good idea. 

```ts
const props = {
  color: getVariant(isDisabled, 'action.background.primary', 'default'),
  borderColor: getVariant(isDisabled, 'action.border.primary', 'default'),
  hoverColor: getVariant(isDisabled, 'action.background.primary', 'hover'),
  hoverBorderColor: getVariant(isDisabled, 'action.border.primary', 'hover')
}
```

Reply:
Oh this looks better https://github.com/razorpay/blade/pull/487/commits/731f8d7a034db03161757d2874c9169110c9a059